packages/xrpl/src/models/methods/submitMultisigned.ts

```diff
@@ -24,28 +25,58 @@ export interface SubmitMultisignedRequest extends BaseRequest {
   fail_hard?: boolean
 }
 
+/**
+ * Common properties for multisigned transaction responses.
+ *
+ * @category Responses
+ */
+interface BaseSubmitMultisignedResult {
+  /**
+   * Code indicating the preliminary result of the transaction, for example.
+   * `tesSUCCESS`.
+   */
+  engine_result: string
+  /**
+   * Numeric code indicating the preliminary result of the transaction,
+   * directly correlated to `engine_result`.
+   */
+  engine_result_code: number
+  /** Human-readable explanation of the preliminary transaction result. */
+  engine_result_message: string
+  /** The complete transaction in hex string format. */
+  tx_blob: string
+  /** The complete transaction in JSON format. */
+  tx_json: Transaction
+}
+
 /**
  * Response expected from a {@link SubmitMultisignedRequest}.
  *
  * @category Responses
  */
 export interface SubmitMultisignedResponse extends BaseResponse {
-  result: {
-    /**
-     * Code indicating the preliminary result of the transaction, for example.
-     * `tesSUCCESS` .
-     */
-    engine_result: string
-    /**
-     * Numeric code indicating the preliminary result of the transaction,
-     * directly correlated to `engine_result`.
-     */
-    engine_result_code: number
-    /** Human-readable explanation of the preliminary transaction result. */
-    engine_result_message: string
-    /** The complete transaction in hex string format. */
-    tx_blob: string
-    /** The complete transaction in JSON format. */
+  result: BaseSubmitMultisignedResult & {
+    hash?: string
+  }
+}
+
+/**
+ * Response expected from a {@link SubmitMultisignedRequest} using api_version 1.
+ *
+ * @category Responses
+ */
+export interface SubmitMultisignedV1Response extends BaseResponse {
+  result: BaseSubmitMultisignedResult & {
```

Comment:
Hey Omar, is my understanding correct? 
`BaseSubmitMultisignedResult` already contains a `tx_json` field, but we are inserting another `tx_json` field as a child property in this line.

I suspect that the javascript runtime flexibly manages the layout of the object, depending on whether its a V1-response or a V2-response, right?

Reply:
It's the property location of `hash` that makes it different between v1 and v2. In `SubmitMultisignedV1Response`, we're overwriting `tx_json` to include `hash` as a child prop.

Reply:
hmm right, makes sense 👍